dask_ml/model_selection/_incremental.py

```diff
@@ -635,7 +643,10 @@ async def _fit(self, X, y, **fit_params):
             context = dummy_context()
 
         X, y, scorer = self._validate_parameters(X, y)
-        X_train, X_test, y_train, y_test = self._get_train_test_split(X, y)
+
+        X_train, X_test, y_train, y_test = self._get_train_test_split(
+            X, y, shuffle=True
```

Comment:
Is this `shuffle=True` needed? There's made a modification above where `shuffle=True` is hard-coded.

Reply:
This method warns when given a dataframe and `shuffle=` is not specified.  We need to choose some default.  Would `False` be a better choice?

Reply:
Ah, I see. I prefer `shuffle=True` because it's the right choice from an optimization perspective. For example, what if the Dask Dataframe are CSVs from each state? Calling `partial_fit` on any one CSV is not the right answer if each CSV is very different.

Reply:
I think that in this function shuffle is only per-partitions, not between partitions.  I think that we would leave inter-partition shuffling to the user ahead of sending to Hyperband

Reply:
> I think that in this function shuffle is only per-partitions, not between partitions.

That's not true; it looks like for default value of `blockwise` depends on whether it's a Dask Array or DataFrame: https://github.com/dask/dask-ml/blob/fc6a04aec4517abca318bc7e44e99f8e540b6fe6/dask_ml/model_selection/_split.py#L392-L395

But that's unrelated to this PR. I think `shuffle=True` is a good choice in case there's order. Maybe we should make `_get_train_test_split` public?

Reply:
> Maybe we should make _get_train_test_split public?

What's the motivation for that?

---

I think

Reply:
Sorry, my last comment is misleading. DataFrame's shuffle_split is definately blockwise-only. So I think we have everything we want here? `shuffle=True, blockwise=True` was the default behavior on master, and that's still what we have?

Reply:
>> Maybe we should make _get_train_test_split public?
>
> What's the motivation for that?

In case someone wants a specific train/test split. Maybe they have time series and don't want `shuffle=True` or `blockwise=True`? Or maybe they want the test set to be a specific chunk from the Dask Array?

Reply:
> shuffle=True, blockwise=True was the default behavior on master, and that's still what we have?

`shuffle=True` is definitely the behavior on master; `shuffle=False` isn't supported ([_split.py#L493][1], [_split.py#L477][2]), and `blockwise` is False for DataFrames, and True for Arrays. I think that's the right behavior.

I need to investigate `blockwise` some; I'm not sure if there's a mixup in the documentation or implementation.

[1]:https://github.com/dask/dask-ml/blob/fc6a04aec4517abca318bc7e44e99f8e540b6fe6/dask_ml/model_selection/_split.py#L493
[2]:https://github.com/dask/dask-ml/blob/fc6a04aec4517abca318bc7e44e99f8e540b6fe6/dask_ml/model_selection/_split.py#L477

Reply:
I think blockwise is backwards for DataFrame in the documentation & implementation. It only supports the equivalent of `blockwise=True` (within-block shuffling).

Reply:
> I think blockwise is backwards for DataFrame in the documentation & implementation. It only supports the equivalent of blockwise=True (within-block shuffling).

I'm a little confused. Should the implementation support shuffling between chunks or within chunks? I'd like to see `train_test_split` support shuffling between chunks (`blockwise=False`). Currently, `train_test_split` only supports shuffling within chunks.

Reply:
> Should the implementation support shuffling between chunks or within chunks? 

Ideally yes. But that's distinct from this PR.